src/EnergyPlus/Construction.cc

```diff
@@ -2063,4 +2062,113 @@ void ConstructionProps::setArraysBasedOnMaxSolidWinLayers(EnergyPlusData &state)
+    std::string key = this->getCacheKey(state);
+    state.dataCache->cache[Cache::CTFKey][key] = cacheData;
+}
+
+void ConstructionProps::loadFromCache(EnergyPlusData &state)
+{
+    try {
+        // load cached data
+        std::string key = this->getCacheKey(state);
+
+        // get the cached data if it exists. if it doesn't nlohmann::json throws out_of_range
+        nlohmann::json thisConstrData = state.dataCache->unorderedCTFObjects.at(key);
+
+        // nlohmann::json does some weird things when directly accessing strings
+        // explicitly setting these as string objects here so we can do a proper comparison
+        std::string a = thisConstrData.at("full_data_key");
+        std::string b = this->getCacheKeyString(state);
+
+        // final confirmation that the object we found matches exactly with the current construction
+        if (a != b) {
+            // show warning message here?
+            this->CTFLoadedFromCache = false;
+            return;
+        }
+
+        // we can load the data if if we've made it this far
+
+        // CTF arrays
+        Cache::jsonToArray(state, this->CTFCross, thisConstrData, "ctf_cross");
+        Cache::jsonToArray1(state, this->CTFFlux, thisConstrData, "ctf_flux");
+        Cache::jsonToArray(state, this->CTFInside, thisConstrData, "ctf_inside");
+        Cache::jsonToArray(state, this->CTFOutside, thisConstrData, "ctf_outside");
+
+        // other necessary data
+        Cache::jsonToData(state, this->NumHistories, thisConstrData, "num_histories");
+        Cache::jsonToData(state, this->NumCTFTerms, thisConstrData, "num_ctf_terms");
+        Cache::jsonToData(state, this->CTFTimeStep, thisConstrData, "ctf_timestep");
+        Cache::jsonToData(state, this->UValue, thisConstrData, "u_value");
+
+        this->CTFLoadedFromCache = true;
+    } catch (const nlohmann::json::out_of_range &e) {
+        // should already be defaulted to false, but this makes sure
+        this->CTFLoadedFromCache = false;
+    }
+}
+
+std::string ConstructionProps::getCacheKey(EnergyPlusData &state)
+{
+    unsigned long long key = 0;
+
+    constexpr int precision_bits(15);
+
+    // data from material layers
+    // adding the layer num to get a unique key for objects with reversed layers
+    for (int Layer = 1; Layer <= this->TotLayers; ++Layer) {
+        int CurrentLayer = this->LayerPoint(Layer);
+        auto &mat = state.dataMaterial->Material(CurrentLayer);
+        key ^= Cache::prepFloatForCacheKey(mat.Thickness, precision_bits) << Layer;
+        key ^= Cache::prepFloatForCacheKey(mat.Conductivity, precision_bits) << Layer;
+        key ^= Cache::prepFloatForCacheKey(mat.Density, precision_bits) << Layer;
+        key ^= Cache::prepFloatForCacheKey(mat.SpecHeat, precision_bits) << Layer;
+        key ^= Cache::prepFloatForCacheKey(mat.Resistance, precision_bits) << Layer;
+    }
+
+    return format("{}", key);
+}
+
+std::string ConstructionProps::getCacheKeyString(EnergyPlusData &state)
```

Comment:
New construction member method with all material layer properties for a final check to verify we've found the right cached data.